src/Cardano/AddressV3.cpp

```diff
@@ -33,14 +66,16 @@ bool AddressV3::parseAndCheckV3(const std::string& addr, NetworkId& networkId, K
         if (!success) {
             return false;
         }
-        if (conv.size() != EncodedSize) {
+
+        if (!parseAndCheckV3(conv, networkId, kind, bytes)) {
+            return false;
+        }
+
+        // check prefix
+        if (const auto expectedHrp = getHrp(kind); addr.substr(0, expectedHrp.length()) != expectedHrp) {
```

Comment:
Now I'm even thinking if we can use:  https://en.cppreference.com/w/cpp/string/basic_string/starts_with since we are in c++20

Reply:
This is a nice improvement, we can do it in another pr